### DISCUSSION TITLE
Unknown Interfaces and Unknown Class

### DISCUSSION THREAD

**Discussion by rhift** [symfony/symfony#48786] [Q&A] [answered]
### Symfony version(s) affected

6.2.2

### Description

Hey there,

I think I accidentally found some references that do not exist. Mostly unknown interfaces, but also one unknown parent. They are all in the HttpClient Component.
```
Warning: Can't preload unlinked class Symfony\Component\HttpClient\Internal\AmpResolver: Unknown interface Amp\Dns\Resolver in vendor/symfony/http-client/Internal/AmpResolver.php on line 26
Warning: Can't preload unlinked class Symfony\Component\HttpClient\Internal\AmpListener: Unknown interface Amp\Http\Client\EventListener in vendor/symfony/http-client/Internal/AmpListener.php on line 26
Warning: Can't preload unlinked class Symfony\Component\HttpClient\Psr18Client: Unknown interface Psr\Http\Client\ClientInterface in vendor/symfony/http-client/Psr18Client.php on line 56
Warning: Can't preload unlinked class Symfony\Component\HttpClient\Psr18NetworkException: Unknown interface Psr\Http\Client\NetworkExceptionInterface in vendor/symfony/http-client/Psr18Client.php on line 216
Warning: Can't preload unlinked class Symfony\Component\HttpClient\Psr18RequestException: Unknown interface Psr\Http\Client\RequestExceptionInterface in vendor/symfony/http-client/Psr18Client.php on line 235
Warning: Can't preload unlinked class Symfony\Component\HttpClient\DataCollector\HttpClientDataCollector: Unknown parent Symfony\Component\HttpKernel\DataCollector\DataCollector in vendor/symfony/http-client/DataCollector/HttpClientDataCollector.php on line 25
Warning: Can't preload unlinked class Symfony\Component\HttpClient\Internal\AmpBody: Unknown interface Amp\Http\Client\RequestBody in vendor/symfony/http-client/Internal/AmpBody.php on line 26
Warning: Can't preload unlinked class Amp\Socket\Connector@anonymous: Unknown interface Amp\Socket\Connector in vendor/symfony/http-client/Internal/AmpClientState.php on line 147
Warning: Can't preload unlinked class Symfony\Component\HttpClient\Response\HttplugPromise: Unknown interface Http\Promise\Promise in vendor/symfony/http-client/Response/HttplugPromise.php on line 24
Warning: Can't preload unlinked class Symfony\Component\HttpClient\HttplugClient: Unknown interface Http\Client\HttpClient in vendor/symfony/http-client/HttplugClient.php on line 67
Warning: Can't preload unlinked class Symfony\Component\HttpClient\DependencyInjection\HttpClientPass: Unknown interface Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface in vendor/symfony/http-client/DependencyInjection/HttpClientPass.php on line 20
```

In addition to PHP not being able to find them I could not either. However, it is entirely possible I am not aware of some clever engineering where these references are actually useful.

I'm happy to remove them and submit a pull request. Let me know how I can help or why I'm wrong.

Thank you!

### How to reproduce

You can search for the unknown interfaces and unknown parent and they should not be present.

### Possible Solution

Remove them if they actually do not exist or aren't in some other way useful.

**Comment by derrabus** [comment]
What kind of problem do these references cause for you?

**Comment by wouterj** [comment, marked as answer]
Hi! Thanks for filling in a bug report :)

These classes are only used together with optional dependencies. E.g. if you want to use the HttplugClient of the HttpClient component, you must first install the httplug library (upon which these missing interfaces will be available).

This sounds to me like you're preloading too many classes (also unused ones). If you use the Symfony framework, I would recommend using the [`config/preload.php`](https://symfony.com/blog/php-preloading-and-symfony) file instead (this will include only the used classes). Otherwise, you would need to revisit your preloading script to avoid loading these classes.

In any case, with this in mind this sounds more like a community support than a bug report to me, so I'm converting it into a "discussion".

**Comment by rhift** [comment]
Ah, they exist somewhere! Thanks @wouterj !

**Comment by rhift** [comment]
Feel free to close this!
